en/codegen/layout-codegen.md

```diff
@@ -25,43 +25,112 @@ In the below image the developer would select the Sketch Groups `peopleList` and
+
+The fluid layout is in a tree structure. The branches and the root are rows and columns, the leafs are components.
+
+## Heuristics and rules
+In order to be properly adjust the elements in HTML, the Code generation applies heuristics and rules to get responsive design from statically positioned elements.
+
+<img class="responsive-img" src="../images/layout_codegen_responsive.gif" />
+
+> [!Note]
+> Conversion to fluid design can not guarantee pixel perfect outcome or HTML and CSS as a developer would write it. For some designs the components or generated groups(rows/columns) may appear displaced. The users should do a review of the generated code and apply changes if needed.
+
+> [!Note]
+> If the designer wants to ensure some components to be part of of one group(row or column), he should put them in a group in Sketch. Groups in Sketch will be respected.
+
+### Navigation menu components
+The Navbar, NavDrawer, Bottom Navigation require special handling. These elements typically represent root level menus. When building the layout these special case components are "moved" at the end of the group, this will make them appear on top of other elements(if there are overlaps).
 
+### Overlaps
+Code generation finds elements that overlap each other. Elements coming partially or fully on top of others are also removed from further heuristics and are given absolute positioning with appropriate z-index. Also overlapping elements are grouped together to ensure the upper element will overlap the element bellow into the new fluid layout as it was in the design from Sketch. Typical example of this would an avatar with a badge.
 
-## Dialog, Toast, Snackbar
+<img class="responsive-img" src="../images/layout_codegen_overlap.png" />
 
-The Dialog, Toast, and Snackbar all have one thing in common, they are all shown in the UI dynamically using some form of TypeScript.
+### Backgrounds
+Elements (e.g. shapes) that map to containers (e.g. div-s) that fully contain other elements become their parents. Additional rules are applied to ensure this rule covers only scenarios that have these shapes used as containers. Currently only simple shapes(rectangles and ovals) without images are used for this rule.
+
+<img class="responsive-img" src="../images/layout_codegen_background.png" />
+
+### Pinned elements
+Sketch allows pinning elements to their parents and Code generation handles pinned elements in a specific way.
+  - Having right or bottom pin to an element applies absolute position and the according margin to that element.
+  - Left or top pins are ignored and those elements are included to the layout groups as if they don't have pins.
+  - Opposite pins(left and right/top and bottom) are discarded and not taken into account. For example left and right pins to an element will be handled the same way as if the element doesn't have pins. The idea here is to stretch the element not to apply margins from both opposite sides.
```

Comment:
The assumption behind the rule is that elements with opposite pins should be stretched instead of having margins applied from those sides.